src/solver/infeasible-problem-analysis/unfeasible-pb-analyzer.cpp

```diff
@@ -0,0 +1,49 @@
+#include <fstream>
+#include <algorithm>
+
+#include "unfeasible-pb-analyzer.h"
+#include "variables-bounds-consistency.h"
+#include "constraint-slack-analysis.h"
+#include <antares/logs/logs.h>
+
+
+using namespace operations_research;
+
+namespace Antares::Optimization
+{
+
+UnfeasiblePbAnalyzer::UnfeasiblePbAnalyzer(std::vector<std::shared_ptr<UnfeasibilityAnalysis>> analysisList)
+        : analysisList_(analysisList)
+{}
+
+void UnfeasiblePbAnalyzer::run(MPSolver* problem)
+{
+    logs.info();
+    logs.info() << "Solver: Starting unfeasibility analysis...";
+
+    for (auto& analysis : analysisList_)
+    {
+        logs.info();
+        logs.info() << analysis->title() << " : running...";
+        analysis->run(problem);
+        if (analysis->hasDetectedInfeasibilityCause())
+            return;
+
+        logs.notice() << analysis->title() << " : nothing detected.";
+    }
+}
+
+void UnfeasiblePbAnalyzer::printReport()
```

Comment:
if it's just about "printing" things then we can const this method

Reply:
You're right. Done.